rails/app/controllers/external_activities_controller.rb

```diff
@@ -327,8 +327,71 @@ def copy
     end
   end
 
+  def edit_collections
+    authorize @external_activity
+
+    @collections = MaterialsCollection.includes(:materials_collection_items).order(:name).all
+
+    @material = [@external_activity]
+    @assigned_collections = @collections.select{|c| _collection_has_materials(c, @material) }
+
+    @unassigned_collections = @collections - @assigned_collections
+
+    respond_to do |format|
+      format.html # edit_collections.html.haml
+    end
+  end
+
+  def update_collections
+    authorize @external_activity
+
+    @external_activity = ExternalActivity.find(params[:material_id])
+    material_type = params[:material_type]
+    collection_ids = params[:materials_collection_id] || []
+
+    if material_type != "ExternalActivity"
+      raise "Unsupported type: #{material_type}"
+    end
+
+    if collection_ids.count > 0
+      add_material_to_collection(collection_ids, @external_activity, material_type)
+      flash[:notice] = "#{@external_activity.name} is assigned to the selected collection(s) successfully."
+    else
+      flash[:error] = "Select at least one collection to assign this #{@external_activity.material_type.classify}"
+    end
+
+    redirect_to action: 'edit_collections', material_id: @external_activity.id, material_type: material_type
+  end
+
   private
 
+  def add_material_to_collection(collection_ids, material, material_type)
```

Comment:
Since this is model based code, it would be better to move it to a model file. 
I think it'd make sense to put it in the ExternalActivity model.

Then you can call `@external_activity.add_to_collections(collection_ids)`